packages/blobs/src/store.ts

```diff
@@ -105,11 +108,51 @@ export type WriteResult = {
   modified: boolean
 }
 
+function otel<This extends Store, Args extends any[], Return>(getAttributes?: (...args: Args) => Attributes) {
+  return function (method: (...args: Args) => Promise<Return>, context: ClassMethodDecoratorContext) {
+    const methodName = String(context.name)
+    const operationName = `blobs.${methodName}`
+    return async function (this: This, ...args: Args): Promise<Return> {
+      const storeName = this['name']
+      const tracer = await getTracer()
+
+      if (tracer) {
+        return tracer.withActiveSpan(operationName, async (span) => {
+          span.setAttribute('blobs.store', storeName)
+          if (getAttributes) {
+            for (const [name, value] of Object.entries(getAttributes(...args))) {
+              value && span.setAttribute(`blobs.${name}`, value)
+            }
+          }
+          try {
+            this['span'] = span
+            const result = await method.apply(this, args)
+            span.setStatus({ code: SpanStatusCode.OK })
+            return result
+          } catch (error) {
+            const message = (error as Error).message ?? 'problem'
+            span.setStatus({
+              code: SpanStatusCode.ERROR,
+              message,
+            })
+            span.recordException(error as Error)
+            throw error
```

Comment:
`SugaredTracer` we get from `@netlify/otel` will already automatically handle recording errors ( https://github.com/open-telemetry/opentelemetry-js/blob/41ba7f57cbf5ae22290168188b467e0c60cd4765/api/src/experimental/trace/SugaredTracer.ts#L19-L24 )

Reply:
oh neat, thank u